docs/hyperion-setup.md

```diff
@@ -1 +1,411 @@
+
+Assuming that your elasticsearch instance is running on localhost and requires no auth, we can use the following bare configuration for the data source (a full list of options can be found [here](/datasource/plugins/elasticsearch/README.md)). Put the following in `elasticsearch.yml`:
+
+```yaml
+poll-interval: 5
+
+elasticsearch:
+    hostname: localhost
+    port: 9200
+    index: logs  
+    scheme: http
+    timestamp-field: "@timestamp"
+    authentication: false
+    response-hit-count: 10000
+
+pipeline:
+    manager-host: "localhost:3000"
+    plugin-id: "ElasticSearch"
+```
+
+As you can see, we have configured the pipeline manager host to point to the port 3000 that we configured in step 0. If we were to start this plugin right now though, it'd error. Why? Because the plugin manager doesn't know anything about the `ElasticSearch` plugin yet and therefore can't tell it where the data needs to go. Let's fix that first. Edit your `pluginmanager.yml` such that instead of it having an empty `plugins` section, it instead reads the following:
+
+```yaml
+plugins:
+  - id: "ElasticSearch"
+    host: "tcp://localhost:3001"
+```
+
+Make sure to restart your plugin manager, then start the elastic search data source as follows:
+
+```shell script
+$ java -jar path/to/elasticsearch-all.jar run elasticsearch.yml
+```
+
+If everything goes according to plan, the ElasticSearch data source will start broadcasting log data on port 3001. Now, let's add some plugins.
+
+### Step pre-2: What data are we actually processing?
+
+We're now able to get data into the Hyperion pipeline. As per the introduction, we eventually need to transform it into a format that our aggregator can understand. To do that, it might be a good idea to have a look at what data we currently have, so that we can set up a plan on how to transform it into the format that the aggregator requires.
+
+The ElasticSearch data source simply pushes the documents stored in ES in a standard JSON format. As such, let's have a look at what kind of data is stored in ES. You can inspect the data by either going to Kibana and expanding a log entry, or by looking at the [LogStash configuration directly](https://github.com/nickyu42/mock-elastic-logging/blob/master/logstash.conf).
+
+Below is a screenshot of Kibana, with some of the more interesting fields highlighted:
+
+![](https://i.imgur.com/qhvFObO.png)
+
+From this screenshot, we know that the data produced by our data source will look something like this:
+
+```json
+{
+    "@timestamp": "2020-06-02T10:03:44.000Z",
+    "log4j_file": "com.sap.enterprises.server.impl.TransportationService",
+    "log4j_level": "INFO",
+    "log4j_line": "11",
+    "log4j_message": "transportation service created with id=5.088856004746587",
+    // extra fields...
```

Comment:
It might be cleaner to change this to something like:
`"extra fields": "..."`
so it doesn't produce the error in highlighting